docs/adr/0004-bundling-experiments.md

```diff
@@ -0,0 +1,74 @@
+
+## Decision Drivers
+
+* Availability of experiments early on the first run
+* No impact on experimentation data analysis
+* Flexibility in creating experiments
+* Ability to quickly disable experiments
+* Simplicity of releases
+
+## Considered Options
+
+* **(A) Do Nothing**
+    * Keep everything the way it is, preventing us from experimenting on users early on their first run
+* **(B) Bundle Experiment data with app on release**
+    * On release, have an `initial_experiments.json` that defines the experiments that will be applied early on the first run
+    * Later on the first run, the client would retrieve the actual experiment data from remote-settings and overwrite the bundled data
+* **(C) Retrieve Experiment data on first run, and deal with delay**
+    * We can retrieve the experiment data on the first run, experiment data however will not be available until after a short delay (network I/O + some disk I/O)
+
+## Decision Outcome
+
+The proposed option: **(B) Bundle Experiment data with app on release** was **rejected**
+
+The option was rejected mainly due to difficulty in disabling experiments and pausing enrollments. This can create a negative user experience as it prevents us from disabling any problematic experiments. Additionally, it ties experiment creation with application release cycles.
+
+Moving forward, we are discussing **(C) Retrieve Experiment data on first run, and deal with delay** to address the first product experimentation requirement, which involves the ability to run experiments on a user's first run, but not necessarily early (The first experiments don't happen to involve onboarding)
+
+### Positive Consequences
+
+* Since the experiments are bundled with the app, they will always be available on first run, without having to wait on network I/O. This allows us to create experiments early on the user's first run (for example, onboarding experiments)
+
+### Negative Consequences
+
+* Inability to disable experiments without issuing a dot-release. The hinders our ability to quickly rollback any experiments that are causing problems. As a consequence, this will slow us down in that we would not be able to launch experiments and iterate quickly if it goes wrong
+* Inability to pause enrollments remotely. We currently have a mechanism to pause enrollments directly on experimenter, however, since the experiment data from remote-settings is not applied until the second run, users would still get enrolled in experiments (whose enrollment has been paused) on their first run.
+* Experiment creation will be tied to release cycles, any experiments that would like to target users on their first run would need to be created before a release is made. This removes flexibility in experiment creation and creates limitation on the timeline experiments can be created to be tied with release cycles (or would force us to create dot-releases for experiments)
```

Comment:
This is probably entirely my misunderstanding of this section of the ADR - (my first one to write 😅). I approached this section as a description of the negative consequences of the main option in discussion (which was bundling for this ADR). I can restructure the ADR so that the main point of this ADR is to propose the alternative we actually went with (as opposed to rejecting the one we didn't go with) and I _think_ it'll make more sense